File: bsep_backend/src/main/java/com/example/bsep_backend/pki/service/CSRService.java
package com.example.bsep_backend.pki.service;

import com.example.bsep_backend.domain.User;
import com.example.bsep_backend.pki.domain.CSRStatus;
import com.example.bsep_backend.pki.dto.CSRResponse;
import com.example.bsep_backend.pki.dto.CreateCSRRequest;
import com.example.bsep_backend.pki.dto.ReviewCSRRequest;

import java.util.List;

public interface CSRService {

    CSRResponse createCSR(CreateCSRRequest request, User requester);

    List<CSRResponse> getMyCSRs(User requester);

    List<CSRResponse> getAllCSRs();

    List<CSRResponse> getCSRsByStatus(CSRStatus status);

    CSRResponse reviewCSR(Long csrId, ReviewCSRRequest request, User reviewer);

    CSRResponse getCSRById(Long csrId);
}

File: bsep-frontend/src/services/csrService.ts
import api from '../api/api';
import { CreateCSRRequest, CSRResponse, ReviewCSRRequest, CSRStatus, UploadCSRRequest } from '../types/csr';

class CSRService {
  async createCSR(request: CreateCSRRequest): Promise<CSRResponse> {
    const response = await api.post('/csr', request);
    return response.data;
  }

  async uploadCSR(request: UploadCSRRequest): Promise<CSRResponse> {
    const formData = new FormData();
    formData.append('csrFile', request.csrFile);
    formData.append('validityDays', request.validityDays.toString());

    const response = await api.post('/csr/upload', formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    });
    return response.data;
  }

  async getMyCSRs(): Promise<CSRResponse[]> {
    const response = await api.get('/csr/my');
    return response.data;
  }

  async getAllCSRs(): Promise<CSRResponse[]> {
    const response = await api.get('/csr');
    return response.data;
  }

  async getCSRsByStatus(status: CSRStatus): Promise<CSRResponse[]> {
    const response = await api.get(`/csr/status/${status}`);
    return response.data;
  }

  async getCSRById(id: number): Promise<CSRResponse> {
    const response = await api.get(`/csr/${id}`);
    return response.data;
  }

  async reviewCSR(id: number, request: ReviewCSRRequest): Promise<CSRResponse> {
    const response = await api.post(`/csr/${id}/review`, request);
    return response.data;
  }
}

export const csrService = new CSRService();
